plugins/out_prometheus_remote_write/remote_write_conf.c

```diff
@@ -189,6 +196,119 @@ struct prometheus_remote_write_context *flb_prometheus_remote_write_context_crea
     /* Set instance flags into upstream */
     flb_output_upstream_set(ctx->u, ins);
 
+
+#ifdef FLB_HAVE_AWS
+    /* AWS Auth */
+    ctx->has_aws_auth = FLB_FALSE;
+    tmp = flb_output_get_property("aws_auth", ins);
```

Comment:
I would advocate for using **flb_managed_chain_provider_create()** instead copying and pasting this section of code. In the HTTP plugin, when we added signv4 there was introduced a new initialization function called: flb_managed_chain_provider_create(). If you switch over, it could make maintenance easier.

This manages the variables that are needed for AWS signv4. 

Configuration managed: https://github.com/fluent/fluent-bit/blob/master/plugins/out_http/http_conf.c#L117-L148
Variables managed: https://github.com/fluent/fluent-bit/blob/master/plugins/out_http/http.h#L41-L49

Reply:
It assumes you have
```
"<config_key_prefix>region"
"<config_key_prefix>sts_endpoint"
"<config_key_prefix>role_arn"
"<config_key_prefix>external_id"
```
In your config map